EIPS/eip-5170.md

```diff
@@ -0,0 +1,152 @@
+
+The authenticity of the client script may be obtained through the `scriptURI()` function call, as in [EIP-5169](./eip-5169.md), or supplied separately by the use-cases. However, this ERC is applicable to any code or data that is signed, and a client must validate the signature in the way specified in this ERC. In real-life use-cases, the client scripts can be either supplied by the aforementioned `scriptURI()` or offered to the client (wallet) in any way the wallet can work, even through NFC connections or QR codes.
+
+### Implementation
+
+Some possible methods of implementation: Note: we cannot deliver a completely off-line solution unless a hosting app can pre-verify the *smart contract deployment key* address. 
+
+1. Simplest (assuming internet connection and written *after* publication of this EIP):
+
+- JWS object outlined below (containing: The script itself, URI to certificate, signature of the script signed by *script signing key*) is linked to via scriptURI() in the contract.
+- 1: determine *smart contract deployment key*. Contract should preferably implement standard 'Ownable' interface or at least the 'owner()' function.
+- 2: fetch the JWS object by querying the `scriptURI()` in the contract.
+- 3: fetch the certificate linked to from the JWS in the `x5u` header.
+- 4: validate that certificate is signed by the *smart contract deployment key*.
+- 5: obtain the address of the *script signing key* from the `SubjectPublicKeyInfo` field of the certificate.
+- 6: obtain the script itself; in our example it will be bundled in the JWS object itself, along with the signature attached to the script, in the `<ds:SignatureValue>` tag.
+- 7: verify that the signature from the `<ds:SignatureValue>` tag is the correct signature of the keccak'd script by the *script signing key*.
+- 8: current time is within `notBefore` and `notAfter`.
+- 9: if the *script signing key* is still valid according to the certificate *and* the script has been signed correctly by the same key, allow user to interact with the token contract(s) defined in the script via the script interface.
+
+2. Example script applied to contract published before this EIP:
+
+- User scans NFC beacon on their mobile.
+- Obtains payload with an app intent (eg ticketing app) containing a URL to a JWS object.
+- JWS Object is fetched along with the script pointed to from the JWS object.
+- Determine origin contract from script.
+- Obtain the owner of the script query `owner()` or JWS object provides contract creation transaction the hosting app can validate and obtain *smart contract deployment key* address.
+- Continue from `3:` above, skipping `6:` as we already have the script.
+- If the *script signing key* is still valid according to the certificate *and* the script has been signed correctly by the same key, allow user to interact with the token contract(s) defined in the script via the script interface.
+
+### Specification
```

Comment:
ERCs usually specify an interface or behaviors in the specifications and then provide some reference implementation. Can you update the specification and implementation sections accordingly?